tests/plugin-parser/unittest_mlops_plugin_parser.cc

```diff
@@ -0,0 +1,383 @@
+  MOCK_METHOD (int, pkgmgrinfo_pkginfo_get_root_path, (pkgmgrinfo_pkginfo_h handle, char **root_path));
+  MOCK_METHOD (int, pkgmgrinfo_pkginfo_get_res_type, (pkgmgrinfo_pkginfo_h handle, char **res_type));
+  MOCK_METHOD (int, pkgmgrinfo_pkginfo_get_res_version, (pkgmgrinfo_pkginfo_h handle, char **res_version));
+};
+
+PkgMgrInfoMock *pkgMgrInfoMockInstance = nullptr;
+
+/**
+ * @brief Mock for pkgmgrinfo_pkginfo_get_pkgid
+ */
+extern "C"
+int pkgmgrinfo_pkginfo_get_pkginfo (const char *pkgid, pkgmgrinfo_pkginfo_h *return_handle)
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_get_pkginfo (pkgid, return_handle);
+}
+
+/**
+ * @brief Mock for pkgmgrinfo_pkginfo_destroy_pkginfo
+ */
+extern "C"
+int pkgmgrinfo_pkginfo_destroy_pkginfo (pkgmgrinfo_pkginfo_h handle)
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_destroy_pkginfo (handle);
+}
+
+/**
+ * @brief Mock for pkgmgrinfo_pkginfo_get_type
+ */
+extern "C"
+int pkgmgrinfo_pkginfo_get_type (pkgmgrinfo_pkginfo_h handle, char **return_type)
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_get_type (handle, return_type);
+}
+
+/**
+ * @brief Mock for pkgmgrinfo_pkginfo_get_root_path
+ */
+extern "C"
+int pkgmgrinfo_pkginfo_get_root_path (pkgmgrinfo_pkginfo_h handle, char **return_path)
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_get_root_path (handle, return_path);
+}
+
+/**
+ * @brief Mock for pkgmgrinfo_pkginfo_get_res_type
+ */
+extern "C"
+int pkgmgrinfo_pkginfo_get_res_type (pkgmgrinfo_pkginfo_h handle, char **return_res_type)
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_get_res_type (handle, return_res_type);
+}
+
+/**
+ * @brief Mock for pkgmgrinfo_pkginfo_get_res_version
+ */
+extern "C"
+int pkgmgrinfo_pkginfo_get_res_version (pkgmgrinfo_pkginfo_h handle, char **return_res_version)
+{
+  return pkgMgrInfoMockInstance->pkgmgrinfo_pkginfo_get_res_version (handle, return_res_version);
+}
+
+/**
+ * @brief pluging-parser test fixture
+ */
+class PkgMgrInfoMockTestFixture : public ::testing::Test
+{
+  protected:
+  static GModule *module;
+  static GTestDBus *dbus;
+
+  /**
+   * @brief Set up test suite.
+   */
+  static void SetUpTestSuite ()
+  {
+    g_autofree gchar *current_dir = g_get_current_dir ();
+    g_autofree gchar *services_dir
+        = g_build_filename (current_dir, "tests", "services", NULL);
+
+    dbus = g_test_dbus_new (G_TEST_DBUS_NONE);
+    ASSERT_TRUE (dbus != nullptr);
+
+    g_test_dbus_add_service_dir (dbus, services_dir);
+    g_test_dbus_up (dbus);
+
+    module = g_module_open ("plugin-parser/libmlops-plugin-parser.so", (GModuleFlags) G_MODULE_BIND_LAZY);
+    ASSERT_TRUE (module != nullptr);
+
+    pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
+  }
+
+  /**
+   * @brief Tear down test suite.
+   */
+  static void TearDownTestSuite ()
+  {
+    delete pkgMgrInfoMockInstance;
+    g_module_close (module);
+    if (dbus) {
```

Comment:
Q: minor question, is checking nullptr (dbus) necessary?
then need check for module.
```
if (module) {
  g_module_close (module);
  module = nullppr;
}
```
or
```
g_module_close (module);
g_test_dbus_down (dbus);
g_object_unref (dbus);
```

Reply:
The latter would be enough. I adopt it. Thanks!